src/blocks/FlashingTextBlock.tsx

```diff
@@ -0,0 +1,138 @@
+import Block from './Block'
+import { BlockModel } from './types'
+import BlockFactory from './BlockFactory';
+import './BlockStyles.css'
+import Box from "@mui/material/Box";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import { useState, useEffect } from "react";
+
+interface FlashingTextProps {
+  content: string;
+  contentColor: string;
+  backgroundColor: string;
+}
+
+function FlashingText({ content, contentColor, backgroundColor } : FlashingTextProps) {
+  const [colors, setColors] = useState({textColor: contentColor, bgColor: backgroundColor});
+  const { textColor, bgColor }  = colors;
+
+  /*
+  This conditional is important when changing themes. If we only had setColors() without the surrounding
+  conditional, then the component crashes because React thinks it's too many rerenders. The conditional
+  makes sure that when there is a new color combination, the state changes to reflect that
+  */
+  if ((contentColor !== textColor && contentColor !== bgColor) || 
+      (backgroundColor !== textColor && backgroundColor !== bgColor)) {
+    setColors({textColor: contentColor, bgColor: backgroundColor})
+  }
+
+  // swap colors after the component renders. changing the state triggers another render,
+  // so there's an infinite loop of color swaps (we trick react)
+  useEffect(() => {
+    function swapColors() {
+      setColors({textColor: bgColor, bgColor: textColor});
+    }
+
+    const timeoutId = setTimeout(swapColors, 1000);
+    return () => clearTimeout(timeoutId);
+  }, [bgColor, textColor]);
+  
+
+  let backgroundStyles = {
+    backgroundColor: bgColor,
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    width: "100%",
+    height: "100%"
+  } as React.CSSProperties
+
+  let textStyles = {
+    color: textColor,
+    backgroundColor: bgColor,
+    textAlign: "center",
+    fontSize: "64px",
+    overflowWrap: "break-word"
+  } as React.CSSProperties
+  
+  return (
+    <div style={backgroundStyles}>
+      <h1 style={textStyles}>
+        {content}
+      </h1>
+    </div>
+  )
+}
+
+export default class FlashingTextBlock extends Block {
+
+  render() {
+    
+    if (Object.keys(this.model.data).length === 0) {
+      return BlockFactory.renderEmptyState(this.model, this.onEditCallback!);
+    }
+
+    let text = this.model.data['text'];
+
+    console.log("render")
```

Comment:
for submission we can delete this (but I liked in testing that I could see it doesn't render too often!)